process-phoenix/src/main/java/com/jakewharton/processphoenix/ProcessPhoenix.java

```diff
@@ -59,7 +59,7 @@ public static void triggerRebirth(Context context, Intent nextIntent) {
     intent.addFlags(FLAG_ACTIVITY_NEW_TASK); // In case we are called with non-Activity context.
     intent.putExtra(KEY_RESTART_INTENT, nextIntent);
     context.startActivity(intent);
-
+    ((Activity) context).finish();
```

Comment:
You still need the `instanceof` check here otherwise being called with a non-Activity context will crash. Perhaps we should force it being called with one.